tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -0,0 +1,383 @@
+                }
+                else
+                {
+                    span.ResourceName = activity.OperationName;
+                }
+            }
+
+            // Update Type with a reasonable default
+            if (string.IsNullOrWhiteSpace(span.Type))
+            {
+                span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
+            }
+        }
+
+        internal static void SetTagObject(Span span, string key, object? value)
+        {
+            if (value is null)
+            {
+                span.SetTag(key, null);
+                return;
+            }
+
+            switch (value)
+            {
+                case char c: // TODO: Can't get here from OTEL API, test with Activity API
+                    AgentSetOtlpTag(span, key, c.ToString());
+                    break;
+                case string s:
+                    AgentSetOtlpTag(span, key, s);
+                    break;
+                case bool b:
+                    AgentSetOtlpTag(span, key, b ? "true" : "false");
+                    break;
+                case byte b: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, b);
+                    break;
+                case sbyte sb: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, sb);
+                    break;
+                case short sh: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, sh);
+                    break;
+                case ushort us: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, us);
+                    break;
+                case int i: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, i);
+                    break;
+                case uint ui: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, ui);
+                    break;
+                case long l: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, l);
+                    break;
+                case ulong ul: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, ul);
+                    break;
+                case float f: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, f);
+                    break;
+                case double d:
+                    span.SetMetric(key, d);
+                    break;
+                case Array array:
+                    AgentSetOtlpTag(span, key, ComposeArrayString(array));
+                    break;
+                default:
+                    AgentSetOtlpTag(span, key, value.ToString()!);
```

Comment:
We might want to check if value implements `IFormattable.ToString(format: null, CultureInfo.InvariantCulture)` and use that if possible.

Reply:
Hmm I'm not sure if it would be a good idea to pass in a `null` value, even if `IFormattable` implementations are supposed to treat that as a "default" case. Since we're not going to be handling numerical types here, I think it would be safer to just call `ToString()` and let the type handle the formatting

Reply:
Sounds good since we're handling all the numeric types in their own switch cases above. Thanks!